### ISSUE TITLE
Add Support for neo4j 2.2.0

### ISSUE THREAD

**Issue by wwwdata** [jmcvetta/neoism#71]
Hi, it seems that neoism does not connect to neo4j 2.2.0,I always get the error

```
<*errors.errorString | 0xc20802ab00>: {
            s: "Invalid database.  Check URI.",
        }
```

however seems to work fine with 2.1.x versions

according to the manual, we need to send an Authorization header now with a base64 encoded user:password string http://neo4j.com/docs/2.2.0/rest-api-security.html

but it is still possible to deactivate authentication. So that is a workaround for now: http://neo4j.com/docs/2.2.0/security-server.html#security-server-auth

**Comment by tolomaus** [comment]
I dont think the fix is needed, just specify http(s)://neo4j:password@server/user/neo4j as the url for the neo4j server and it works

**Comment by wwwdata** [comment]
you are right, go automatically extract user and password from the url and sets the basic auth header :P i will remove my user/password stuff and squash the commits. Then we have one env variable for the NEO4J_URL for testing, and the tests are working again. I think that is worth merging too

**Comment by tolomaus** [comment]
Are you connecting over https?
I have temporarily added the following lines to skip the cert check:

```golang
	tr := &http.Transport{
		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
	}
	client := &http.Client{Transport: tr}
```
But it would be good to allow passing a cert in. Just started with go but if I fix it I will send a PR.

**Comment by wwwdata** [comment]
hm, I never tried it via https. But it makes sense to add an option to skip the cert check for that

**Comment by wwwdata** [comment]
I updates my PR #72 with some test fixes and travis build with neo4j 2.2.0. I will keep that open, because it makes sense to merge it because it fixes the tests. I will close this ticket now because 2.2.0 works and we don't have to change the library for it. @tolomaus you should open a new issue for the https cert skip problem.

**Comment by tolomaus** [comment]
Sure no problem
